app/V1Module/presenters/AssignmentSolutionsPresenter.php

```diff
@@ -264,7 +264,71 @@ public function actionSetBonusPoints(string $id) {
     $this->sendSuccessResponse("OK");
   }
 
-  public function checkSetAcceptedSubmission(string $id) {
+  public function checkSetFlag(string $id) {
+    $solution = $this->assignmentSolutions->findOrThrow($id);
+    if (!$this->assignmentSolutionAcl->canSetFlag($solution)) {
+      throw new ForbiddenRequestException("You cannot change flags for this submission");
+    }
+  }
+
+  /**
+   * Set flag of the assignment solution.
+   * @POST
+   * @param string $id identifier of the solution
+   * @param string $flag name of the flag which should to be changed
+   * @Param(type="post", name="value", required=true, validation=boolean, description="True or false which should be set to given flag name")
+   * @throws NotFoundException
+   * @throws \Nette\Application\AbortException
+   * @throws \Exception
+   */
+  public function actionSetFlag(string $id, string $flag) {
+    $req = $this->getRequest();
+    $solution = $this->assignmentSolutions->findOrThrow($id);
+    if ($solution->getAssignment() === null) {
+      throw new NotFoundException("Assignment for solution '$id' was deleted");
+    }
+
+    if ($solution->getSolution()->getAuthor() === null) {
+      throw new NotFoundException("Author of solution '$id' was deleted");
+    }
+
+    // map of boolean flag names with the information about uniqueness
+    $knownBoolFlags = [
+      "accepted" => true,
+      "reviewed" => false
+    ];
+
+    if (!array_key_exists($flag, $knownBoolFlags)) {
+      throw new BadRequestException("Trying to set unknown boolean flag '$flag' to the solution");
+    }
+
+    // handle given flag
+    $unique = $knownBoolFlags[$flag];
+    $value = filter_var($req->getPost("value"), FILTER_VALIDATE_BOOLEAN);
+    // handle unique flags
+    if ($unique && $value) {
+      // flag has to be set to false for all other submissions
```

Comment:
Maybe a little bit more accurate here. That is all *solutions* of a user to an assignment (not all as all-*all*).

Reply:
Will do